Livecheckables/isc-dhcp.rb

```diff
@@ -1,6 +1,6 @@
 class IscDhcp
   livecheck do
     url "https://www.isc.org/downloads/"
-    regex(%r{/dhcp-([a-zA-Z0-9.]+)\.t}i)
+    regex(%r{/dhcp-v?(\d+(?:\.\d+)+)\.t}i)
```

Comment:
```suggestion
    regex(%r{/dhcp-v?(\d+(?:\.\d+)+(?:-P\d+)?)\.t}i)
```

Looking at the [release notes](https://gitlab.isc.org/isc-projects/dhcp/-/blob/master/RELNOTES), the releases with letters are typically alpha, beta, and release candidates, so I'm good with omitting those.

However, ISC DHCP does sometimes have patch releases like `4.3.6-P1` and these have been used in the formula in the past, so I think we should continue to match those.